### ISSUE TITLE
Question: please explain the score more in depth

### ISSUE THREAD

**Issue by g-pavlik** [DmitryTsepelev/rubocop_director#10]
I love this gem!

To sell this to my team I need to understand how the scores are calculated, and I have problems with that. 

`<number of offences> * <cop weight> * <number of file updates> * <update weight>`

Some of the values are self-explanatory. For example: the more often the file is updated - the higher score. 
Default cop weights are 1, so we can ignore those (and they are subjectively set for each project). 

`<number of offences>`  - why does the number of offenses for a particular cop should increase the score of a specific file? Or did I get it all wrong?

`<update weight>` - what is that?

**Comment by DmitryTsepelev** [comment]
Hey! Sounds like I'll need to update docs after we discuss it and maybe make formula more granular.

`<number of offences>` is how many times we see a particular cop offence in the file, more offences — more weight (but we can influence the importance of the cop by `<cop weight>`)
`<update weight>` is used to influence the importance of how many times the file was updated

**Example**

We have two files:

- app/models/user.rb (Rails/SomeCop once and Rails/AnotherCop once) updated 5 times
- app/controller/user_controller.rb (Rails/SomeCop twice) updated 2 times

Config:

```
"default_cop_weight" => 1,
"update_weight" => 1,
"weights" => {
  "Rails/SomeCop" => 1,
  "Rails/AnotherCop" => 1
}
```

Result:

```
{:path=>"app/models/user.rb", :updates_count=>5, :offense_counts=>{"Rails/SomeCop"=>1, "Rails/AnotherCop"=>1}, :value=>10.0}
{:path=>"app/controller/user_controller.rb", :updates_count=>2, :offense_counts=>{"Rails/SomeCop"=>2}, :value=>4.0}
```

Let's say we care less about AnotherCop (0.5)

```
{:path=>"app/models/user.rb", :updates_count=>5, :offense_counts=>{"Rails/SomeCop"=>1, "Rails/AnotherCop"=>1}, :value=>7.5}
{:path=>"app/controller/user_controller.rb", :updates_count=>2, :offense_counts=>{"Rails/SomeCop"=>2}, :value=>4.0}
```

Let's say we care less about updates (0.1):

```
{:path=>"app/models/user.rb", :updates_count=>5, :offense_counts=>{"Rails/SomeCop"=>1, "Rails/AnotherCop"=>1}, :value=>1.0}
{:path=>"app/controller/user_controller.rb", :updates_count=>2, :offense_counts=>{"Rails/SomeCop"=>2}, :value=>0.4}
```

Maybe we shoud sum instead of multiply? (i.e. `for_each_cop(<number of offences> * <cop weight>) + <number of file updates> * <update weight>`) Otherwise we cannot say we don't care about updates by using 0, it will make total weighth 0 🤔

**Comment by g-pavlik** [comment]
> I love this gem!
> 
> To sell this to my team I need to understand how the scores are calculated, and I have problems with that.
> 
> `<number of offences> * <cop weight> * <number of file updates> * <update weight>`
> 
> Some of the values are self-explanatory. For example: the more often the file is updated - the higher score. Default cop weights are 1, so we can ignore those (and they are subjectively set for each project).
> 
> `<number of offences>` - why does the number of offenses for a particular cop should increase the score of a specific file? Or did I get it all wrong?
> 
> `<update weight>` - what is that?

I don't know about that. Maybe you're right. 

What I was looking for is a conceptual approach: 
- the more a file is updated - the more beneficial it is to fix its offenses
- the more offenses it has in total - the more beneficial it is to fix its offenses

Is this how you think about it?

**Comment by DmitryTsepelev** [comment]
I've just rewritten the algorithm https://github.com/DmitryTsepelev/rubocop_director/pull/13

> the more offenses it has in total - the more beneficial it is to fix its offenses

This is how it worked and this is how it still works: more offences give more weight (multiplied by importance — for instance wrong indent is less terrifying than nested transactions).

> the more offenses it has in total - the more beneficial it is to fix its offenses

I did something similar. First of all we find how many updates we had for all files. After that, for each file we do:

1. calculate sum of offences multiplied by weights (`<cop value>`)
2. find how often the file was relatively updated (`<count of file updates> / <total count of updates>`)
3. get the power of <update weight> for the value above (`<update importance>`)
4. result value is `<cop value> * <update importance>`

It should allow us to configure the importance of updates. For instance, imagine two files with the cop value 100, first one was updated 5 times and second — 15 times.

With update weight = 1.5 (we do care about updates):

```
File 1: 100 * (5/(5+15)) ** 1.5 = 12.5
File 2: 100 * (15/(5+15)) ** 1.5 = 65
```

With update weight = 0.1 (we do not care about updates):

```
File 1: 100 * (5/(5+15)) ** 0.1 = 87
File 2: 100 * (15/(5+15)) ** 0.1 = 97
```
